src/python/pants/backend/python/tasks/pytest_run.py

```diff
@@ -112,6 +112,13 @@ def register_options(cls, register):
              help='Emit coverage information for specified packages or directories (absolute or '
                   'relative to the build root).  The special value "auto" indicates that Pants '
                   'should attempt to deduce which packages to emit coverage for.')
+    # TODO: The default is False for backwards compatiblity. We almost certainly want to change
+    # that after a deprecation cycle.
```

Comment:
I recommend that we start the deprecation now in this PR. How would we do this? If the default is used but not explicitly specified, log a `DeprecationWarning`?

Reply:
Hm, after rereading this PR, in an ideal world would this even be an option? Can you imagine use cases where someone would want this to be `False`?

I know we have to consider backwards compatibility, but want to make sure in two release cycles that this option still would make sense to have.

Reply:
We could just treat this as a bug and change it without deprecation, if people are on board with that.

Reply:
We could just treat this as a bug and change it without deprecation, if people are on board with that.

Reply:
I think this is a bug - we didn’t seem to intentionally implement this behavior and I see no case where someone would want it. I support changing without deprecation and without adding an option.

Reply:
ping @jsirois @stuhood @asherf  (this comment in particular, and the PR in general)

Reply:
I don't have enough context on this so I don't have an informed option if this is a bug or a deprecation.

Reply:
Yep, fine with that.

Will defer to @Eric-Arellano  for the rest.